drivers/iommu/riscv/iommu.h

```diff
@@ -26,12 +28,27 @@
 struct riscv_iommu_queue {
 	dma_addr_t base_dma;
 	void *base;
-	u32 len;		/* single item length */
+	size_t len;		/* single item length */
 	u32 cnt;		/* items count */
 	u32 lui;		/* last used index, consumer/producer share */
-	u32 qbr;		/* queue base register offset */
-	u32 qcr;		/* queue control and status register offset */
+	unsigned qbr;		/* queue base register offset */
+	unsigned qcr;		/* queue control and status register offset */
 	int irq;		/* registered interrupt number */
+	bool in_iomem;		/* indicates queue data are in I/O memory  */
+};
+
+enum riscv_queue_ids {
+	RISCV_IOMMU_COMMAND_QUEUE = 0,
+	RISCV_IOMMU_FAULT_QUEUE = 1,
+	RISCV_IOMMU_PREQ_QUEUE = 2
```

Comment:
nit: we could use `RISCV_IOMMU_PAGE_REQUEST_QUEUE` or `RISCV_IOMMU_PAGE_REQ_QUEUE`.
Or one alternative is to simplify all of them:
```c
enum riscv_queue_ids {
	RISCV_IOMMU_CQ = 0,
	RISCV_IOMMU_FQ = 1,
	RISCV_IOMMU_PQ = 2
}
```

Reply:
I'll go with RISCV_IOMMU_PAGE_REQUEST_QUEUE